service.go

```diff
@@ -229,15 +236,27 @@ func (hs *k8sHealthcheckService) addAck(serviceName string, ackMessage string) e
 	return nil
 }
 
-func (hs *k8sHealthcheckService) getDeployments() (map[string]deployment, error) {
+func (hs *k8sHealthcheckService) getDeployments() (deployments map[string]deployment, err error) {
 	deploymentList, err := hs.k8sClient.ExtensionsV1beta1().Deployments(namespace).List(v1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve deployments: %v", err.Error())
 	}
 
-	deployments := make(map[string]deployment)
+	deployments = make(map[string]deployment)
 	for _, d := range deploymentList.Items {
-		deployments[d.Name] = deployment{
+		deployments[d.GetName()] = deployment{
+			desiredReplicas: *d.Spec.Replicas,
+		}
+	}
+
+	dl, err := hs.k8sClient.AppsV1beta1().StatefulSets(namespace).List(v1.ListOptions{})
+	if err != nil {
+		log.WithError(err).Warn("error getting StatefulSet list")
+		return deployments, nil
```

Comment:
Should we not treat this as an error? If we can't get the statefulsets, we won't see any sign of it on the healthcheck page right?

Reply:
Good point. I was attempting to not break previous behaviour but you are right. If we fail here it should return an error as well.